source/val/validate_memory.cpp

```diff
@@ -390,6 +396,23 @@ spv_result_t CheckMemoryAccess(ValidationState_t& _, const Instruction* inst,
              << "Memory accesses Aligned operand value " << aligned_value
              << " is not a power of two.";
     }
+
+    uint32_t largest_scalar = 0;
+    if (dst_sc == spv::StorageClass::PhysicalStorageBuffer) {
+      largest_scalar =
+          _.GetLargestScalarType(dst_pointer_type->GetOperandAs<uint32_t>(2));
```

Comment:
This won't work with untyped pointers.

Reply:
...ya, will get some up

so if I have the following conceptual code

```
layout(buffer_reference) buffer B {
    uint x;
};

layout(set=0, binding=0) buffer SSBO {
    B b;
};

void main() {
    b.x = 0;
}
```

There are 2 `OpTypePointer`, one to the `StorageBuffer` and one to the `PhysicalStorageBuffer` 

I assume the only issue I need to worry about is an untyped pointer of `PhysicalStorageBuffer` since this code is caught on the `OpLoad`/`OpStore` already

Reply:
Actually trying this out, not sure how a `OpTypeUntypedPointerKHR PhysicalStorageBuffer` would work here, there is no `PhysicalStorageBuffer` Variable to pass the `Base` operand to for the untyped Access Chain

Reply:
awwwwww, ok, I see, the issue is `StorageBuffer` is now an untyped variable ... think I got it now